doc/TROUBLESHOOT.md

```diff
@@ -101,6 +101,35 @@ Otherwise, the reference to the first non-needed object on the path (`staticX`,
 staticX -> A -> B -> disposedD
 ```
 
+Be careful not to fall into pattern to limit your fix to just releasing the reference to the disposed object. 
+Instead, you need to release just most left object that is not needed any more (first check `staticX`, then `A`, then `B`). All objects
+referenced from it will also become unreachable, and thus available for garbage collection.
+
+Remember, not all leaks are detected, but for 
+[the leak tracked objects]([staticX -> A -> B](https://github.com/dart-lang/leak_tracker/blob/main/doc/DETECT.md#by-tracked-classes)) 
+only. So, if you just release the 
+reference to disposed object, you will fix the detected leak and hide the undetected ones.
+
+And, there is no point to release reference to child inside the method `dispose` of the parent, because
+not needed parent should be released together with its disposal. If
+your fix for a leak is like this, you are defenitely hiding the leak: 
```

Comment:
I'm not sure this is true.

If the child is not nulled out, it may not be eligible for garbage collection (even if it has released some resource in dispose). And if it is _not_ nulled out, other code in the class that depends on its nullability to determine whether it can be used or not may fail.

For example, it's pretty common in Flutter code to mark some native object as nullable, and once it is disposed null it out so it doesn't get used again.

Reply:
I am not saying it is wrong to null out. I am saying it is wrong to null out in `dispose`.
If parent is disposed, it should become non-used, unreachable and thus to be garbage collected together with child.

Another side effect of nulling in dispose is that you have to make the field nullable, that adds unnecessary complexity to the code and eliminates non-nullable optimizations.

How does it sound?

Reply:
I see. There are pros and cons nulling out member in dispose.

However, the statement here is: a leak fix should not be nulling out in `dispose`, because, while it makes the leak smaller, it does not fix it.

This means it is ok to null out the member in dispose in a separate PR, but not in order to fox the leak. Am I missing something?

Reply:
Yeah, I think pointing out that nulling it out isn't the right way to fix the leak, but I would avoid calling that style bad - it's used frequently in the flutter framework for the reasons I mentioned

Reply:
Modified. Is it better now?